src/app/components/jsx-helpers/page-loader.js

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import {fetchFromCMS} from '~/helpers/controller/cms-mixin';
+import $ from '~/helpers/$';
+
+export default function usePageData(slug, preserveWrapping=false) {
+    const [data, setData] = React.useState();
+
+    React.useEffect(() => {
+        fetchFromCMS(slug, preserveWrapping).then($.camelCaseKeys).then(setData);
+    }, [slug, preserveWrapping]);
```

Comment:
you might consider adding a `debounce` to the `fetchFromCMS` function.  If the props change rapidly you wouldn't want to fire 1,000's of network requests

Reply:
This is for loading a page. I don't expect those props to change. But it does occur to me that I might use different tools to set this up so that it's more clear. 🤔